registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package io.mosip.registration.processor.status.service.impl;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.mosip.kernel.core.dataaccess.exception.DataAccessLayerException;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.registration.processor.core.constant.LoggerFileConstant;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.status.entity.AnonymousProfileEntity;
+import io.mosip.registration.processor.status.entity.AnonymousProfilePKEntity;
+import io.mosip.registration.processor.status.exception.TablenotAccessibleException;
+import io.mosip.registration.processor.status.repositary.BaseRegProcRepository;
+import io.mosip.registration.processor.status.service.AnonymousProfileService;
+@Component
+public class AnonymousProfileServiceImpl implements AnonymousProfileService {
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(AnonymousProfileServiceImpl.class);
+	/** The Anonymus Profile repository. */
+	@Autowired
+	private BaseRegProcRepository<AnonymousProfileEntity, String> anonymousProfileRepository;
+
+
+	@Override
+	public void saveAnonymousProfile(String regId, String processStage, String profileJson) {
+		
+			regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), regId,
+					"PacketInfoManagerImpl::saveAnonymousProfile()::entry");
+			try {
+			AnonymousProfileEntity anonymousProfileEntity=new AnonymousProfileEntity();
+			AnonymousProfilePKEntity anonymousProfilePKEntity=new AnonymousProfilePKEntity();
+			anonymousProfilePKEntity.setId(regId);

```

Comment:
id should not be regid. its random uuid. use -
RegistrationUtility.generateId() no need to pass regid in this method

Reply:
I have changed the code.